modules/codegen/src/main/scala/com/twilio/guardrail/generators/CirceProtocolGenerator.scala

```diff
@@ -158,12 +164,24 @@ object CirceProtocolGenerator {
           } yield ProtocolParameter(term, name, dep, readOnlyKey, emptyToNullKey)
         }
 
-      case RenderDTOClass(clsName, terms) =>
-        Target.pure(q"""
-          case class ${Type.Name(clsName)}(..${terms})
-        """)
+      case RenderDTOClass(clsName, selfTerms, parents) =>
+        val discriminator = parents.find(_.discriminator.isDefined).flatMap(_.discriminator) //collectFirst { case SuperClass(_, _, _, Some(discriminator)) => discriminator }
+        val parentNameOpt = parents.headOption.map(_.clsName)
+        val terms = (parents.reverse.flatMap(_.params.map(_.term)) ++ selfTerms).filterNot(
+          param => discriminator.contains(param.name.value)
+        )
+        val code = parentNameOpt
+          .fold(q"""case class ${Type.Name(clsName)}(..${terms})""")(
+            parentName => q"""case class ${Type.Name(clsName)}(..${terms}) extends ${template"${init"${Type.Name(parentName)}(...$Nil)"}"}"""
```

Comment:
Is it necessary to do this `${template"${init"${Type.Name` stuff? It seems as though `extends ${Type.Name("foo")}(...${Nil})` actually works:
```
scala> q"case class Foo(..${List(param"blop: Blop")}) extends ${Type.Name("blep")}(...${List(List(Term.Name("blop")))})"
res12: meta.Defn.Class = case class Foo(blop: Blop) extends blep(blop)
```

Reply:
Thanks for it.